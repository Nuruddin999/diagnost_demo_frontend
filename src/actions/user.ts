import { createAction } from "@reduxjs/toolkit";
export const TYPES = {
  userLoginType: 'user/login',
  userRegisterType: 'user/register',
  userCheckType: 'user/check',
  userLogOut: 'user/logout',
  userCheckIsAdmin: 'user/checkIsSuperAdmn',
  userGetOne: 'user/getone',
  changeIsDeletedPlaceType: 'user/changeIsDeletedPlaceType',
  getByLetter: 'user/getByLetter',
  userDel: 'user/deleteone',
  userUpdateRights: 'user/updateRights',
  userUpdateSignFile:'user/signupdate',
  userUpdatePrimary:'user/updateprimary'
}
export type registeredUser = {
  email: string,
  password: string,
  name: string,
  phone: string,
  speciality: string,
  role: string,
  files:any[] | null
}

export const login = createAction(TYPES.userLoginType, function prepare(email: string, password: string) {
  return {
    payload: {
      email,
      password
    },
  }
})

export const changeIsDeletedPlaceAction = createAction(TYPES.changeIsDeletedPlaceType, function prepare(email: string) {
  return {
    payload: {
      email,
    },
  }
})


export const registerUser = createAction(TYPES.userRegisterType, function prepare(body: registeredUser) {
  return {
    payload: {
      ...body
    },
  }
})

export const updateRightsAction = createAction(TYPES.userUpdateRights, function prepare(entity: string, field: string, value: boolean, userId: string) {
  return {
    payload: {
      entity, field, value, userId
    },
  }
})
export const updatePrimaryData = createAction(TYPES.userUpdatePrimary, function prepare(email: string, speciality: string, phone: string) {
  return {
    payload: {
      email, speciality, phone
    },
  }
})
export const getUserByLetter = createAction(TYPES.getByLetter, function prepare(page: number,
  limit: number,
  email: string,
  name: string,
  speciality: string,
  phone: string) {
  return {
    payload: {
      page, limit, email, name, speciality, phone
    },
  }
})
export const deleteUser = createAction(TYPES.userDel, function prepare(id: string) {
  return {
    payload: {
      id
    },
  }
})

export const userSignFileUpdate = createAction(TYPES.userUpdateSignFile, function prepare(id: string, files:Array<File>) {
  return {
    payload: {
      id,
      files
    },
  }
})

export const checkUser = createAction(TYPES.userCheckType)
export const logOutUser = createAction(TYPES.userLogOut)
export const checkIsSuperAdmin = createAction(TYPES.userCheckIsAdmin)