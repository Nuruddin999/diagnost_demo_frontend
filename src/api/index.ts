import axios from "axios";

export const diagnostApi = axios.create({
  baseURL: process.env.REACT_APP_BASIC_URL,
});