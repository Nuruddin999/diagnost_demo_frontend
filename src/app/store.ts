import { Action, combineReducers, configureStore, ThunkAction } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import applicationItemSlice from '../reducers/applicationItemSlice';
import applicationSlice from '../reducers/applicationSlice';
import uiSlice  from '../reducers/ui';
import userSlice from '../reducers/userSlice';
import runSagas from '../sagas';

export const rootReducer = combineReducers({
  user: userSlice,
  application:applicationSlice,
  applicationItem:applicationItemSlice,
  ui: uiSlice
})

export type RootState = ReturnType<typeof rootReducer>

const sagaMiddleware = createSagaMiddleware()

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        // Ignore these action types
        ignoredActions: ['listitem/getlistitem'],
      },
    }).concat(sagaMiddleware),
});

sagaMiddleware.run(runSagas)

export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;