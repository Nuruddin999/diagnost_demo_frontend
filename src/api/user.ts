import { registeredUser } from '../actions/user';
import { diagnostApi } from './index';

export const loginApi = async (email: string, password: string) => {
  const response = await diagnostApi.post('/login', {
    email, password
  })
  return response.data
}
export const registerApi = async (body: registeredUser) => {
  const response = await diagnostApi.post('/registration', {
    ...body
  })
  return response.data
}
export const checkAuth = async () => {
  const response = await diagnostApi.post('/refresh', { refreshToken: localStorage.getItem('refreshToken') })
  return response.data
}
export const checkHasSuperAdmin = async () => {
  const response = await diagnostApi.get('/superadmn')
  return response.data
}
export const logOut = async () => {
  const response = await diagnostApi.post('/logout', { refreshToken: localStorage.getItem('refreshToken') })
  return response.data
}
export const changeIsDeletedApi = async (email: string) => {
  const response = await diagnostApi.post('/changedel', {
    email
  }, { headers: { 'Authorization': `Bearer ${localStorage.getItem('refreshToken')}` }, })
  return response.data
}
export const getAllUsersApi = async (page: number,
  limit: number,
  email: string,
  name: string,
  speciality: string,
  phone: string) => {
  const response = await diagnostApi.get('/users', {
    headers: { 'Authorization': `Bearer ${localStorage.getItem('refreshToken')}` },
    params: { page, limit, email, name, speciality, phone }
  })
  return response.data
}
/**
 * Удаление одного пользователя по id.
 * @param {number} id Id заключения.
 */

export const deleteUserApi = async (id: string) => {
  const response = await diagnostApi.get(`/userdel/${id}`,
    {
      headers: { 'Authorization': `Bearer ${localStorage.getItem('refreshToken')}` },
    })
  return response.data
}
export const updateRightApi = async (entity: string, field: string, value: boolean, userId: string) => {
  const response = await diagnostApi.post('/rightupd', {
    entity, field, value, userId
  }, { headers: { 'Authorization': `Bearer ${localStorage.getItem('refreshToken')}` }, })
  return response.data
}
export const updatePrimaryApi = async (email: string, speciality: string, phone: string) => {
  const response = await diagnostApi.post('/usrupd', {
    email, speciality, phone
  }, { headers: { 'Authorization': `Bearer ${localStorage.getItem('refreshToken')}` }, })
  return response.data
}

export const upLoadFileApi = async (files: Array<any>, id: string, onUploadProgress?: any) => {
  let formData = new FormData();
  formData.append("file", files[0]);
  formData.append("userid", id);
  const response = await diagnostApi.post('/upload', formData, {
    headers: {
      'Authorization': `Bearer ${localStorage.getItem('refreshToken')}`,
      "Content-Type": "multipart/form-data",
    },
    onUploadProgress
  })
  return response.data
}


export const getFilesList = async (id:string) => {
  const response = await diagnostApi.get(`/files/${id}`,  {
    headers: {
      'Authorization': `Bearer ${localStorage.getItem('refreshToken')}`,
    },
  })
  return response.data
}


