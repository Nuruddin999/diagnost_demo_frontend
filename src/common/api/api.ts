import { diagnostApi } from "../../api"

/**
 * Получение одного элемента по id.
 * @param {number} id Id заключения.
 *  @param {string} url Id заключения.
 */
export const getListItemById = async (id: string, url: string) => {
    const response = await diagnostApi.get(`/${url}/${id}`, {
        headers: { 'Authorization': `Bearer ${localStorage.getItem('refreshToken')}` },
    })
    return response.data
}