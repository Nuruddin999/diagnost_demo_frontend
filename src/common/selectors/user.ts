import { createSelector } from '@reduxjs/toolkit'
import { RootState } from '../../app/store'
import { Right } from '../../reducers/userSlice'

const selectUserItem = (state: RootState) => state.user.useritem
const selectApplicationUser = (state: RootState) => state.user.user
export const selectsUserItemRights = createSelector(selectUserItem, user => {
  const processedRights: Right[] = []
  user.rights?.forEach((item: Right) => {
    if (item.entity === 'applications') {
      processedRights[0] = item
    }
    else if (item.entity === 'users') {
      processedRights[1] = item
    }
    else {
      processedRights[2] = item
    }
  })
  return processedRights
})
export const selectApplicationUserRights = createSelector(selectApplicationUser, user => {
  type rightsObj = {
    applications?: {
      create: boolean,
      read: boolean,
      update: boolean,
      delete: boolean
    }, users?: {
      create: boolean,
      read: boolean,
      update: boolean,
      delete: boolean
    }, checkupPlanPlace?: {
      create: boolean,
      read: boolean,
      update: boolean,
      delete: boolean
    }
  }
  const processedRights: rightsObj = {}
  user.rights?.forEach((item: Right) => {
    processedRights[item.entity as keyof rightsObj] = { create: item.create, read: item.read, update: item.update, delete: item.delete }
  })
  const isApplicationOneRight = processedRights.applications?.create || processedRights.applications?.read || processedRights.applications?.update || processedRights.applications?.update
  const isUsersOneRight = processedRights.users?.create || processedRights.users?.read || processedRights.users?.update || processedRights.users?.update
  return { processedRights, isApplicationOneRight, isUsersOneRight };
})