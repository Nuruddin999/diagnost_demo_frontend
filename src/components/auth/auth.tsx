import { Button, TextField, Typography } from "@mui/material";
import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { Redirect } from "react-router-dom";
import { RootState } from "../../app/store";
import { Loader } from "../loader/loader";
import '../../common/components/registration/style.auth.scss'
import { login } from "../../actions/user";
import { setError } from "../../reducers/ui";

export const Auth = (): React.ReactElement => {
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const { user } = useSelector((state: RootState) => state.user)
  const { isCircular, errorMessage } = useSelector((state: RootState) => state.ui)
  const dispatch = useDispatch()
  const onSubmit = (event: React.SyntheticEvent) => {
    event.preventDefault();
    dispatch(login(email, password))
  }

  useEffect(() => {
    if (errorMessage) {
      setTimeout(() => dispatch(setError('')), 1500)
    }
  }, [errorMessage])

  return <div className={'auth-wrapper'}>
    <div className={"auth-container"}>
      <form onSubmit={(event) => onSubmit(event)} data-testid='loginform'>
        <TextField value={email}
          type='email'
          required
          size='small'
          fullWidth
          placeholder='Email'
          margin='normal'
          onChange={(event) => setEmail(event.target.value)} />
        <TextField
          value={password}
          type='password'
          required
          size='small'
          fullWidth
          placeholder='Пароль'
          margin='normal'
          onChange={(event) => setPassword(event?.target.value)} />

        { errorMessage && <Typography className='error-reg' align='center'>
          {errorMessage}
        </Typography>}
        <Button className='login-button' fullWidth variant='contained' disableElevation type='submit'>
          <Loader title='Войти' isLoading={isCircular} />
        </Button>
      </form>
      {user.role !== '' && <Redirect to='/main/table' />}
    </div>
  </div>
}