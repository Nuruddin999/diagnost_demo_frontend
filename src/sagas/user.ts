import { updatePrimaryApi } from './../api/user';
import { call, put, select, all } from "redux-saga/effects"
import { getUserByLetter } from "../actions/user"
import { checkAuth, loginApi, logOut, registerApi, checkHasSuperAdmin, changeIsDeletedApi, getAllUsersApi, deleteUserApi, updateRightApi, upLoadFileApi, getFilesList } from "../api/user"
import { RootState } from "../app/store"
import { initialRights } from "../constants"
import { setFileUploadStatus, setProgress, setCircular, setError, setStatus, setAddUserStatus, setUserItemStatus } from "../reducers/ui"
import { changeLoadStatus, changeReqStatus, Right, saveIsDeletedPlaceInUser, saveRightsInApplicationUser, saveRightsInUserItem, saveSuperUser, saveUser, saveUserItemSignFileInfo, saveUsers, User } from "../reducers/userSlice"

type loginUserResponse = {
  accessToken: string,
  refreshToken: string,
  user: {
    id: string,
    email: string,
    name: string,
    phone: string,
    speciality: string,
    role: string,
    rights: Array<Right>,
    urlSignPath?: string,
    signFileName?: string,
    isDeletedPlace: boolean
  }
}
type allUsersResponse = {
  count: number,
  rows: Array<User>
}
/**
 * Вход в систему.
 * @param login .
 */
export function* loginUser(login: { type: 'user/login', payload: { email: string, password: string } }) {
  try {
    yield put(setCircular(true))
    const response: loginUserResponse = yield call(loginApi, login.payload.email, login.payload.password)
    const { accessToken, refreshToken, user } = response
    if (response) {
      localStorage.setItem('dtokenn', accessToken)
      localStorage.setItem('refreshToken', refreshToken)
      yield all(
        [put(setCircular(false)),
        put(saveUser({ ...user, rights: user.rights.length > 0 ? user.rights : initialRights, isLoading: false, reqStatus: 'ok', isDeletedPlace: false })),
        ])
    }
  } catch (e: any) {
    yield all(
      [put(setStatus('no')),
      put(setError(e.response ? e.response?.data?.message : 'Произошла ошибка, попробуйте позже')),
      ])
  }
}
function* calculateProgress(loaded: number, total: number) {
  const progress: number = yield put(setProgress(Math.round((100 * loaded) / total)))
  console.log('progress', progress)
  return progress
}
/**
 * Регистрация нов пользователя.
 * @param body .
 */
export function* registerUser(body: { type: 'user/register', payload: { email: string, password: string, name: string, speciality: string, phone: string, role: string, files: Array<any> } }) {
  try {
    yield put(setAddUserStatus('pending'))
    const response: { user: string } = yield call(registerApi, { ...body.payload, files: null })
    if (response) {
      const fileUploadResponse: string = yield call(upLoadFileApi, body.payload.files, response.user)
      if (fileUploadResponse) {
        yield call(getFilesList, response.user)
        yield put(setAddUserStatus('ok'))
        const hasSuperUser: boolean = yield select((state: RootState) => state.user.hasSuperUser)
        if (!hasSuperUser) {
          yield put(saveSuperUser(true))
        }
        yield put(getUserByLetter(1, 10, '', '', '', ''))
      }
    }
  } catch (e: any) {
    yield all(
      [put(setAddUserStatus('no')),
      put(setError(e.response ? e.response?.data?.message : 'Произошла ошибка, попробуйте позже')),
      ])
  }

}
/**
 * Проверяем есть ли токен активности у пользователя
 */
export function* checkUserAuth() {
  try {
    yield put(changeLoadStatus(true))
    const { superAdmin } = yield call(checkHasSuperAdmin)
    if (!superAdmin) { yield put(saveSuperUser(false)) }
    const response: loginUserResponse = yield call(checkAuth)
    const { accessToken, refreshToken, user } = response
    if (response) {
      localStorage.setItem('dtokenn', accessToken)
      localStorage.setItem('refreshToken', refreshToken)
      yield put(saveUser({ ...user, rights: user.rights.length > 0 ? user.rights : initialRights, isLoading: false, reqStatus: 'ok' }))
    }
  } catch (e: any) {
    if (e.response) {
      yield put(changeReqStatus(e.response?.data?.message))
    }
    else {
      yield put(changeReqStatus('Неизвестаня ошибка'))
    }
    yield put(saveUser({ id: '0', name: 'empty', role: '', phone: '', email: '', speciality: '', rights: [], isLoading: false, reqStatus: 'ok', isDeletedPlace: false }))
  }
}

export function* logoutUser() {
  try {
    yield put(changeLoadStatus(true))
    const response: number = yield call(logOut)
    if (response) {
      localStorage.removeItem('dtokenn')
      localStorage.removeItem('refreshToken')
      yield put(saveUser({ id: '0', email: '', role: '', name: 'empty', isLoading: false, reqStatus: 'no', rights: initialRights, phone: '', speciality: '', isDeletedPlace: false }))
      yield put(changeLoadStatus(false))
    }
  } catch (e: any) {
    if (e.response) {
      yield put(changeReqStatus(e.response?.data?.message))
    }
    else {
      yield put(changeReqStatus('Неизвестаня ошибка'))
    }
  }
}

export function* changeIsDeletedPlace(body: { type: 'user/changeIsDeletedPlaceType', payload: { email: string } }) {
  try {
    const response: loginUserResponse = yield call(changeIsDeletedApi, body.payload.email)
    const { user } = response
    if (response) {
      yield put(saveIsDeletedPlaceInUser(user.isDeletedPlace))
    }
  } catch (e: any) {
    if (e.response) {
      yield put(changeReqStatus(e.response?.data?.message))
    }
    else {
      yield put(changeReqStatus('Неизвестаня ошибка'))
    }
  }
}

/**
 * Сага получения списка пользователей.
 * @param addApplication
 */
export function* fetchUser(getUser: { type: 'user/getByLetter', payload: { page: number, limit: number, email: string, name: string, speciality: string, phone: string } }) {
  try {
    yield put(setStatus('pending'))
    const { page, limit, email, name, speciality, phone } = getUser.payload
    const response: allUsersResponse = yield call(getAllUsersApi, page, limit, email, name, speciality, phone)
    if (response) {
      const { rows, count } = response
      yield all([put(setStatus('ok')), put(saveUsers({ users: rows, count }))])
    }
  } catch (e: any) {
    yield all(
      [put(setStatus('no')),
      put(setError('Произошла ошибка, попробуйте позже')),
      ])
  }
}
/**
 * Сага обновления прав пользователя из списка пользователей.
 * @param addApplication
 */
export function* updateUserRights(updateRight: { type: 'user/updateRights', payload: { entity: string, field: string, value: boolean, userId: string } }) {
  try {
    const { entity, field, value, userId } = updateRight.payload
    const response: allUsersResponse = yield call(updateRightApi, entity, field, value, userId)
    if (response) {
      const rights: Array<Right> = yield select((state: RootState) => state.user.useritem.rights)
      const applicationUser: User = yield select((state: RootState) => state.user.user)
      const changedRights = rights.map((entityName) => entity === entityName.entity ? {
        ...entityName,
        [field]: value
      } : entityName)
      yield put(saveRightsInUserItem({ rights: changedRights }))
      if (applicationUser.id === userId) {
        yield put(saveRightsInApplicationUser({ rights: changedRights }))
      }
    }
  } catch (e: any) {
    yield put(setError('Произошла ошибка, попробуйте позже'))
  }
}
/**
 * Сага обновления специальности и телефона пользователя.
 * @param addApplication
 */
 export function* updateUserPrimary(updatePrimary: { type: 'user/updateRights', payload: { email: string, speciality: string, phone: string } }) {
  try {
    yield put(setUserItemStatus('pending'))
    const { email, speciality, phone } = updatePrimary.payload
    const response: allUsersResponse = yield call(updatePrimaryApi, email, speciality, phone)
    if (response) {
      yield put(setUserItemStatus('ok'))
    }
  } catch (e: any) {
    yield put(setUserItemStatus('Произошла ошибка, попробуйте позже'))
  }
}
/**
* Удаление  пользователя.
* @param {Object} delUser .
*/
export function* removeUser(delUser: { type: 'user/deleteone', payload: { id: string } }) {
  try {
    const { id } = delUser.payload
    const response: {} = yield call(deleteUserApi, id)
    if (response) {
      yield put(changeReqStatus('success'))
      yield put(getUserByLetter(1, 10, '', '', '', ''))
    }
  } catch (e: any) {
    if (e.response) {
      yield put(changeReqStatus(e.response?.data?.message))
    }
    else {
      yield put(changeReqStatus('Неизвестаня ошибка'))
    }
  }
}

/*обноаление картинки подписини у пользователя *
* @param {Object} delUser .
*/
export function* updateUserSignFile(updateUserSignFile: { type: 'user/signupdate', payload: { id: string, files: Array<File> } }) {
  try {
    yield put(setFileUploadStatus('pending'))
    const { id, files } = updateUserSignFile.payload
    const response: { urlSignPath: string, signFileName: string } = yield call(upLoadFileApi, files, id, () => { })
    if (response) {
      yield all([put(setFileUploadStatus('success')), put(saveUserItemSignFileInfo({ urlSignPath: response.urlSignPath, signFileName: response.signFileName }))])
    }
  } catch (e: any) {
    yield all([put(setFileUploadStatus('error')), put(setError("Произошла ошибка, попробуйте позже"))])
  }
}
