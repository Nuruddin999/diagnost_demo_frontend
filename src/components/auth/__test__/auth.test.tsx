import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../app/store';
import { Auth } from '../auth';
import {
  BrowserRouter as Router,
} from "react-router-dom";

import { loginApi } from '../../../api/user';

jest.mock('../../../api/index')
jest.mock('../../../api/user')

describe('Auth ', () => {
  const renderAuth = () => (
    <Provider store={store}>
      <Router>
        <Auth />
      </Router>
    </Provider>
  )
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('renders properly', () => {
    const { container } = render(renderAuth())
    expect(container).toMatchSnapshot()
  })

  it('changes email and password properly', async () => {
    const { container, getByPlaceholderText } = render(renderAuth())
    const emailInput = getByPlaceholderText('Email') as HTMLInputElement;
    const passwordInput = getByPlaceholderText('Пароль') as HTMLInputElement;
    fireEvent.change(getByPlaceholderText('Email'), { target: { value: 'testdoc@toctor.ru' } })
    fireEvent.change(getByPlaceholderText('Пароль'), { target: { value: '12345' } })
    expect(emailInput).toBeTruthy()
    expect(emailInput.value).toBe('testdoc@toctor.ru')
    expect(passwordInput.value).toBe('12345')
    expect(passwordInput).toBeTruthy()
    expect(container).toMatchSnapshot()
  })
  it('sends login request', async () => {
    (loginApi as unknown as jest.Mock).mockResolvedValue({
      accessToken: '12345',
      refreshToken: '5678',
      user: {
        id: '12',
        email: 'testdoc@test.ru',
        name: 'doctor',
        phone: '12345678',
        speciality: 'doctor',
        role: 'doctor',
        rights: [{ entity: 'applications', create: true, update: true, read: true, delete: false }, { entity: 'users', create: false, update: false, read: false, delete: false }, { entity: 'checkupPlanPlace', create: false, update: false, read: false, delete: false }],
        isDeletedPlace: false
      }
    })
    const { getByPlaceholderText, getByTestId, findByTestId } = render(renderAuth())
    const emailInput = getByPlaceholderText('Email') as HTMLInputElement;
    const passwordInput = getByPlaceholderText('Пароль') as HTMLInputElement;
    fireEvent.change(emailInput, { target: { value: 'testdoc@toctor.ru' } })
    fireEvent.change(passwordInput, { target: { value: '12345' } })
    expect(emailInput.value).toBe('testdoc@toctor.ru')
    expect(passwordInput.value).toBe('12345')
    fireEvent.submit(getByTestId('loginform'))
    const loaded = await findByTestId('loaded')
    expect(loaded).toBeInTheDocument()
    expect(screen.queryByText('Неизвестаня ошибка')).not.toBeInTheDocument()

  })
  it('failed request', async () => {
    (loginApi as unknown as jest.Mock).mockResolvedValue({})
    const { getByPlaceholderText, getByTestId, findByText } = render(renderAuth())
    const emailInput = getByPlaceholderText('Email') as HTMLInputElement;
    const passwordInput = getByPlaceholderText('Пароль') as HTMLInputElement;
    fireEvent.change(emailInput, { target: { value: 'testdoc@toctor.ru' } })
    fireEvent.change(passwordInput, { target: { value: '12345' } })
    expect(emailInput.value).toBe('testdoc@toctor.ru')
    expect(passwordInput.value).toBe('12345')
    fireEvent.submit(getByTestId('loginform'))
    const loaded = screen.queryByTestId('loaded')
    expect(loaded).not.toBeInTheDocument()
    expect(await findByText('Неизвестаня ошибка')).toBeInTheDocument()
  })
})
