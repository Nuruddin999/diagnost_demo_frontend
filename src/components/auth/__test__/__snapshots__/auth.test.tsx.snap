// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth  changes email and password properly 1`] = `
<div>
  <div
    class="auth-wrapper"
  >
    <div
      class="auth-container"
    >
      <form
        data-testid="loginform"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-3"
              placeholder="Email"
              required=""
              type="email"
              value="testdoc@toctor.ru"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-4"
              placeholder="Пароль"
              required=""
              type="password"
              value="12345"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root login-button css-1y3no4h-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <div
            class="btn-no-loader"
            data-testid="loaded"
          >
            <span
              class="btn-title"
            >
              Войти
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Auth  renders properly 1`] = `
<div>
  <div
    class="auth-wrapper"
  >
    <div
      class="auth-container"
    >
      <form
        data-testid="loginform"
      >
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-1"
              placeholder="Email"
              required=""
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-2"
              placeholder="Пароль"
              required=""
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButtonBase-root login-button css-1y3no4h-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <div
            class="btn-no-loader"
            data-testid="loaded"
          >
            <span
              class="btn-title"
            >
              Войти
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</div>
`;
